dDatabase/Database.cpp

```diff
@@ -30,15 +30,17 @@ void Database::Connect(const string& host, const string& database, const string&
 	// property manually (which the URL parsing fails to do)
 	const std::string UNIX_PROTO = "unix://";
 	const std::string PIPE_PROTO = "pipe://";
+	const std::string TCP_PROTO = "tcp://";
     if (host.find(UNIX_PROTO) == 0) {
 		properties["hostName"] = "unix://localhost";
 		properties["localSocket"] = host.substr(UNIX_PROTO.length()).c_str();
     } else if (host.find(PIPE_PROTO) == 0) {
 		properties["hostName"] = "pipe://localhost";
 		properties["pipe"] = host.substr(PIPE_PROTO.length()).c_str();
     } else {
-		properties["hostName"] = host.c_str();
+		properties["hostName"] = (TCP_PROTO + host).c_str();
```

Comment:
This seems like an unrelated change?

Reply:
Not unrelated.  If you started MariaDB on any port other than 3306, than your server would fail to start.  This change addresses that issue and allows you to specify the port as is mentioned in the new README

Reply:
c.f. https://github.com/DarkflameUniverse/DarkflameServer/blob/481bc037d53d3bab45f06bbcd1cf0cccb49f2418/README.md?plain=1#L186-L188

Reply:
Yeah, sure, but adding `tcp://` to `hostName` seems like a bugged way of doing this, as per https://mariadb.com/docs/server/connect/programming-languages/cpp/connect. Something like `tcp://localhost:3306` is a *Compatibility Syntax Connection URI* according to that doc and we don't even use that parameter to `connect` because of claims it's buggy in the comment next to it.

What I'm getting at, is this is a code change and has different review criteria as a README, if you make a well-intentioned change in here.

Reply:
I'm not entirely convinced as to why using `tcp://` for a tcp connection is bugged, its only bugged for the other two types because we never specify which one is being used to connect.  I will revert this and leave a note in the README to prepend `tcp://` if you plan to launch on a port besides 3306.  Is that ok?

Reply:
Using `tcp` is the right thing, but in the first param of a `connect(uri, params)` call afaict.